roles/btrfsmaintenance/tasks/main.yml

```diff
@@ -0,0 +1,79 @@
+#########################################################################
+# Title:         Community: btrfsmaintenance                            #
+# Author(s):     owine                                                  #
+# URL:           https://github.com/Cloudbox/Community                  #
+# --                                                                    #
+#         Part of the Cloudbox project: https://cloudbox.works          #
+#########################################################################
+#                   GNU General Public License v3.0                     #
+#########################################################################
+---
+- name: "Determine '/' filesystem type"
+  set_fact:
+    root_fstype: "{{ item.fstype }}"
+  when: (item.mount == '/')
+  with_items:
+    - "{{ ansible_mounts }}"
+
+- name: "Determine '/opt' filesystem type"
+  set_fact:
+    opt_fstype: "{{ item.fstype }}"
+  when: (item.mount == '/opt')
+  with_items:
+    - "{{ ansible_mounts }}"
+
+- name: "Setup btrfsmaintenance scripts"
+  block:
+
+  - name: "Clone btrfsmaintenance repo"
+    git:
+      repo: https://github.com/kdave/btrfsmaintenance.git
+      dest: '/opt/btrfsmaintenance'
+      clone: yes
+      version: HEAD
+      force: yes
+    become: yes
+    become_user: "{{ user.name }}"
+
+  - name: "Set btrfsmaintenance directory permissions"
+    file: "path='/opt/btrfsmaintenance' state=directory mode=0775 owner={{ user.name }} group={{ user.name }} recurse=yes"
+
+  - name: "Change mountpoints to 'auto'"
+    lineinfile:
+      path: "/opt/btrfsmaintenance/sysconfig.btrfsmaintenance"
+      regexp: '^BTRFS_TRIM_MOUNTPOINTS\s?='
+      line: 'BTRFS_TRIM_MOUNTPOINTS="auto"'
+      state: present
+
+  - name: "Execute dist-install.sh"
+    become: yes
+    become_user: root
+    shell: dist-install.sh
```

Comment:
When testing this role, it errored and failed here. I assume because it doesn't have the full path to the file so it's trying to run it at /dist-install.sh where it doesn't exist.